lib/dal/tests/integration_test/materialized_views.rs

```diff
@@ -216,16 +229,224 @@ async fn component(ctx: &DalContext) -> Result<()> {
+            }], // expected
+            beta_mv.connections // actual
+        );
+    }
+
+    // Check the charlie MV.
+    {
+        let charlie_mv =
+            dal_materialized_views::incoming_connections::assemble(ctx.clone(), charlie.id())
+                .await?;
+        assert_eq!(
+            charlie.id(),  // expected
+            charlie_mv.id  // actual
+        );
+        assert_eq!(
+            vec![
+                Connection::Prop {
+                    from_component_id: alpha.id().into(),
+                    from_attribute_value_id: alpha_si_name_attribute_value_id,
+                    from_attribute_value_path: alpha_si_name_attribute_value_path,
+                    from_prop_id: alpha_si_name_prop_id,
+                    from_prop_path: alpha_si_name_prop_path.with_replaced_sep_and_prefix("/"),
+                    to_component_id: charlie.id().into(),
+                    to_attribute_value_id: charlie_si_name_attribute_value_id,
+                    to_attribute_value_path: charlie_si_name_attribute_value_path,
+                    to_prop_id: charlie_si_name_prop_id,
+                    to_prop_path: charlie_si_name_prop_path.with_replaced_sep_and_prefix("/"),
+                },
+                Connection::Prop {
+                    from_component_id: beta.id().into(),
+                    from_attribute_value_id: beta_domain_name_attribute_value_id,
+                    from_attribute_value_path: beta_domain_name_attribute_value_path,
+                    from_prop_id: beta_domain_name_prop_id,
+                    from_prop_path: beta_domain_name_prop_path.with_replaced_sep_and_prefix("/"),
+                    to_component_id: charlie.id().into(),
+                    to_attribute_value_id: charlie_domain_name_attribute_value_id,
+                    to_attribute_value_path: charlie_domain_name_attribute_value_path,
+                    to_prop_id: charlie_domain_name_prop_id,
+                    to_prop_path: charlie_domain_name_prop_path.with_replaced_sep_and_prefix("/"),
+                },
+                Connection::Socket {
+                    from_component_id: beta.id().into(),
+                    from_attribute_value_id: beta_output_socket_attribute_value_id,
+                    from_attribute_value_path: beta_output_socket_attribute_value_path,
+                    from_socket_id: beta_output_socket.id(),
+                    from_socket_name: beta_output_socket.name().to_owned(),
+                    to_component_id: charlie.id().into(),
+                    to_socket_id: charlie_input_socket.id(),
+                    to_socket_name: charlie_input_socket.name().to_owned(),
+                    to_attribute_value_id: charlie_input_socket_attribute_value_id,
+                    to_attribute_value_path: charlie_input_socket_attribute_value_path,
+                }
+            ], // expected
+            charlie_mv.connections // actual
```

Comment:
These are sorted on the way out because MV generation requires stable ordering, so we do not need to sort the connections again.